### PR TITLE
refactor the dynamic sidebar links (menuLinks) to be extendable to control all dashboard links

### PR DESCRIPTION
**Pull request by ca-scribner**
This PR refactors the existing dynamic dashboard implementation to make it more easily reused with other dashboard link types.  

changes include:
* refactoring the library to use new nomenclature
* removing some unnecessary dashboard link tests
* refactoring the link handling logic of the charm into separate functions for easier reuse

The general function of the charm shouldn't change, just the implementation.

Notes to reviewers:
* linting wont pass (will be cleaned up in future PR)
* unit tests should pass
* integration tests.... tbd :)

### CODE REVIEW

File: src/dashboard_links.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    DashboardLink,
+    dashboard_links_to_json,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def aggregate_links(
+    links_from_relation: List[DashboardLink], additional_link_config: str, link_order_config: str
+):
```
Comment (NohaIhab) [inline src/dashboard_links.py:18]: docstring missing here, same for the method below

**Review by NohaIhab — COMMENTED** [review]
refactoring looks good, left one small comment

File: src/dashboard_links.py (hunk 1 above)
Comment (ca-scribner) [reply src/dashboard_links.py:18]: added!

**Review by misohu — APPROVED** [review]
LGTM good job

**Pull request merged**